### ISSUE TITLE
Allow PersistentView to Work on Multiple PersistentActor instances

### ISSUE THREAD

**Issue by chanan** [akka/akka#15004]
This is a feature request for akka-persistence. It would be very useful to allow a view to operate on more than one processor. This was discussed here: https://groups.google.com/forum/#!topic/akka-user/dhyZ-WpwPz4

One way would be to have processorId() return a List<String> instead of String. Another way suggested by Patrik in the thread above is to allow to tag processors and have the view watch the tag.

**Comment by enpassant** [comment]
+1

**Comment by smlin** [comment]
Somebody already did PR? Any link to code example?

Tags for DDD aggregate +1

**Comment by ktoso** [comment]
No PR yet, the new read side will use Akka Streams so we're not yet focused on it.
The current persistence effort is around stabilizing the write side, which includes some important things we need to nail there as the plugin APIs will become stable with this release.

**Comment by olger** [comment]
Hi Konrad,

I’ve made some changes to support the aggregateId based on akka persistence in the master branch. 
Basically, it adds an aggregateId that maybe set by the PersistentActor (actually via Eventsourced) in order to support the whole read construct. 
Actually the aggregateId is added to the PersistentRepr as an Option[String] (like the persistenceId as String). 
I did not change the protobuf side (yet) and added a ReplayAggregatedMessages Request that was intended to support the read side. 

Basically, I try to get the support in in a non intrusive way (in line with the current structure and code) but I don’t think that it will match your future plans. 
What would be helpful for you to get the ‘tag’ / aggregate construct in the code ? (with or without the code I am working on) 

Kind regards, 

Olger

**Comment by ktoso** [comment]
Hi Olger!
Thanks for offering to help :-) The next step on the road to this feature is this ticket: https://github.com/akka/akka/issues/17579 
It is a component which will allow, among other things, tagging without changing too many APIs, and also be more useful than "just tagging". 

In essence we want to allow Journals to define on what datatype they work. Then each Journal could provide its specialized type the adapter has to return. For example if your Journal needs tags, that specialized type (part of that Journals SPI basically) would include such fields. And if a Journal does not  need these fields it could use the simple datatypes we have now.

**Comment by olger** [comment]
Great, I’ll have a look and see if I understand what is required. 

Kind regards,

Olger

**Comment by patriknw** [comment]
this is solved by the query api
